crates/apollo-compiler/src/diagnostics.rs

```diff
@@ -400,17 +421,35 @@ impl ApolloDiagnostic {
         } else {
             ReportKind::Error
         };
+
+        let span = self.map_location(self.location);
+
         let mut colors = ColorGenerator::new();
-        let mut builder = Report::build(severity, self.location.file_id(), self.location.offset())
-            .with_message(&self.data);
-        builder.add_labels(
-            self.labels
-                .iter()
-                .map(|label| ariadne::Label::from(label.clone()).with_color(colors.next())),
-        );
+        let mut builder =
+            Report::build(severity, self.location.file_id(), span.1.start).with_message(&self.data);
+        builder.add_labels(self.labels.iter().map(|label| {
+            ariadne::Label::new(self.map_location(label.location))
+                .with_message(&label.text)
+                .with_color(colors.next())
+        }));
         if let Some(help) = &self.help {
             builder = builder.with_help(help);
         }
         builder.finish()
     }
+
+    pub fn to_json(&self) -> GraphQLError {
+        let mut locations = vec![];
+
+        if let Some((line, column)) = self.get_line_column() {
+            locations.push(GraphQLLocation { line, column });
+        }
+
+        GraphQLError {
+            message: self.data.to_string(),
```

Comment:
ok awesome! this should totally get us far in validation swap. i was thinking some time in the future when we are less swamped, we can tweak the messages for serialisable errors even, so the stuff that was written out before with the aid of source code would be more readable on its own.

Reply:
Yeah I have a separate branch for that, mostly adding the names of things to the messages themselves. I think it's actually pretty crucial for landing validation because it's really confusing without.